include/app/tabs/settings.hpp

```diff
@@ -95,4 +97,28 @@ class OpenAutoSettingsSubTab : public QWidget {
     Theme *theme;
 };
 
+class CameraSettingsSubTab : public QWidget {
+    Q_OBJECT
+
+   public:
+    CameraSettingsSubTab(QWidget *parent = nullptr);
+
+   signals:
+    void cam_toggle_requested();
+    void cam_name_changed(QString);
+
+   public slots:
+    void on_newConnectionStatus(QString);
+
+   private slots:
+    void on_camName_editingFinished();
+    void on_streamAddress_editingFinished();
```

Comment:
One of the really nice things about AUTOUIC is that it automatically detects slots and connects them to the corresponding signals with no code. You just add the handler with the correct format and it gets called.

Reply:
so i actually never explicitly defined slots... I prefer using lambdas for all the connections

Reply:
That certainly has its place but I prefer to keep an object's logic in the actual object so if the connect() is in a different object it violates this encapsulation.

Reply:
well if you use the lambda the scope is even tighter because its defined entirely within the connection, that way you don't need to define any functions that only serve a single purpose

Reply:
not really sure how it violates the encapsulation

Reply:
It's a little hard to explain but basically the "business logic" of each module should be self-contained. For example, when someone presses "connect" on the camera settings page the camera module does stuff to manage the stream. If that stuff was in a connect() call in the main window, then code in the main window would have to talk about camera module internals.

When dealing with cross-module communication, the information that is transmitted (in this example) is that the user pressed "connect". The camera module would normally be given the sole authority what to do with that.

I don't know if this is clear at all but generally by separating concerns as much as possible the code is less likely to have bugs and is easier to understand and maintain.

Reply:
not really sure i follow

i agree that anything that needs to be done internally to the camera module should be done solely in that module, but why would using a lambda break that rule? The lambda would just act as the slot, just that you don't need to define a function for that single purpose